packages/flutter/lib/src/material/expansion_panel.dart

```diff
@@ -121,8 +170,80 @@ class ExpansionPanelList extends StatelessWidget {
   /// The duration of the expansion animation.
   final Duration animationDuration;
 
+  // Whether multiple panels can be open simultaneously
+  final bool _allowOnlyOnePanelOpen;
+
+  @override
+  State<StatefulWidget> createState() => new _ExpansionPanelListState();
+}
+
+class _ExpansionPanelListState extends State<ExpansionPanelList> {
+  ExpansionPanelRadio _currentOpenPanel;
+
+  @override
+  void initState() {
+    super.initState();
+    if (widget._allowOnlyOnePanelOpen) {
+      for (ExpansionPanelRadio child in widget.children) {
+        final ExpansionPanelRadio widgetChild = child;
+        final bool expansionPanelValid = _currentOpenPanel == null && widgetChild.initializesExpanded ||
+                                         !widgetChild.initializesExpanded;
+        assert(expansionPanelValid, 'Trying to initialize radio panel list with more than two open panels!');
+        if (widgetChild.initializesExpanded)
+          _currentOpenPanel = widgetChild;
+      }
+    }
+  }
+
+  @override
+  void didUpdateWidget(ExpansionPanelList oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget._allowOnlyOnePanelOpen) {
+      bool panelExpanded = false;
+      for (ExpansionPanelRadio newChild in widget.children) {
+        final ExpansionPanelRadio child = newChild;
+        final int childKey = child.value;
+        final bool panelWillBeExpanded = child.initializesExpanded || (childKey == _currentOpenPanel?.value);
+        assert(!(panelExpanded && panelWillBeExpanded),
+          'Trying to initialize radio panel list with more than two open panels!');
+        panelExpanded = panelWillBeExpanded;
+
+        _currentOpenPanel = childKey == _currentOpenPanel?.value ? child : _currentOpenPanel;
+        if (_currentOpenPanel == null && child.initializesExpanded)
+          _currentOpenPanel = child;
+      }
+    }
+    else if(oldWidget._allowOnlyOnePanelOpen) {
+      _currentOpenPanel = null;
+      setState((){});
```

Comment:
You don't/shouldn't setState in didUpdateWidget, because your state is already dirty.  Might have been a miscommunication - this is only required in _handlePressed